pkg/geo/geomfn/validity_check.go

```diff
@@ -64,6 +67,37 @@ func IsValidDetail(g geo.Geometry, flags int) (ValidDetail, error) {
 	}, nil
 }
 
+// IsValidTrajectory returns whether a geometry encodes a valid trajectory
+func IsValidTrajectory(line geo.Geometry) (bool, error) {
+	if line.ShapeType() != geopb.ShapeType_LineStringM && line.ShapeType() != geopb.ShapeType_LineStringZM {
+		return false, errors.New("geom is not a LINESTRING(Z)M")
+	}
+	t, err := line.AsGeomT()
+	if err != nil {
+		return false, err
+	}
+	if t.Empty() {
+		return false, errors.New("line is an empty LINESTRING(Z)M")
+	}
+
+	isValid := true
+	mIndex := geom.Layout.MIndex(t.Layout())
+	currIndex := mIndex
+	prevIndex := -1
+	flatLen := len(t.FlatCoords())
+
+	for currIndex < flatLen {
```

Comment:
can we use `t.Coords()` for the for loop to make this cleaner, i.e.

```
coords := t.Coords()
for i := 1; i < len(coords); i++ {
   if coords[mIndex][i-1] <= .... {
      return false, nil // avoid the isValid variable by returning directly
   }
}
return true, nil
```

Reply:
Looks like the geom.T interface doesn't have a Coords field. 
I only see FlatCoords in the interface.

Reply:
Looks like the geom.T interface doesn't have a Coords field.
I only see FlatCoords in the interface.

Reply:
since we are asserting linestrings, we can do
```
lineString. ok := t.(*geom.LineString)
if !ok {
  return errors.AssertionFailedf("expected linestring, got %T", t)
}
```

and use `lineString.Coords()`